producer/writer/consumer_writer.go

```diff
@@ -194,6 +204,11 @@ func (w *consumerWriterImpl) resetConnectionUntilClose() {
 	for {
 		select {
 		case <-w.resetCh:
+			// Avoid resetting too frequent.
+			if w.resetTooSoon() {
```

Comment:
Do you want to simply sleep till the target time then? Otherwise if there's a lot of reset request, you end up looping into this same function.

Reply:
sounds good

Reply:
Actually I don't think I will be running into this very frequently, once the write error triggered the start of the reset, the validConn will be set to false, and future writes will fail fast and not try reset, so this path will not be constantly hit. There is a metrics for restTooSoon, we can add back the sleep if we see that metrics being too high.